### ISSUE TITLE
Android remoteNotificationReceived event fired multiple times 

### ISSUE THREAD

**Issue by jbenzshawel** [CatalystCode/react-native-azurenotificationhub#86]
When the application is opened from tapping a push notification the 'remoteNotificationReceived' event is fired multiple times for the same notification.

Example:
1. Open app by pressing push notification in app tray 
2. Remote notification event fires with payload of push notification sent when app closed
3. Send another notification with app open
4. Remote notification event fires with payload of notification sent when app open
5. Put the app in background then foreground 
6. Remote notification event fires again with payload corresponding to push notification sent when the app was closed. 
7. Whenever app is brought active from background the notification event fires again with the payload corresponding to the push notification sent when app was closed. This behavior continues until the app is restarted. 

When adding the event listener I am using the following to ensure the event listener has not been added multiple times: 

```js
const onReceivedListeners = PushNotificationEmitter.listeners(DEVICE_NOTIF_EVENT);

if (onReceivedListeners.length === 0) {
    PushNotificationEmitter.addListener(DEVICE_NOTIF_EVENT, pushNotificationRecevied);
}
```

The app is using the "singleTop" android launch mode.

**Comment by phongcao** [comment]
@jbenzshawel I'll try to have a look.

**Comment by jbenzshawel** [comment]
Thanks! I have this fixed with JS by doing the following but it would be nice if the native code could handle it: 

```js

const androidGoogleMessageIds = [];

const _onPushNotificationReceived = notification => {
    let pushNotification =
        Platform.OS === 'ios' ? notification.getData() : JSON.parse(notification);

    if (Platform.OS === 'android') {
        const googleMessageId = pushNotification['google.message_id'];
        if (androidGoogleMessageIds.some(x => x === googleMessageId)) {
            return;
        } else {
            androidGoogleMessageIds.push(googleMessageId);
        }
    }
   // continue handling notification 
}
```

**Comment by phongcao** [comment]
@jbenzshawel Can you please verify this issue on #94? It should be fixed now.

**Comment by phongcao** [comment]
Feel free to re-open if it happens in the master version.

**Comment by jbenzshawel** [comment]
@phongcao Sounds good! Sorry I haven't had a chance to test this. I should be able to this weekend. Has a new NPM version been released with this fix?

**Comment by phongcao** [comment]
No problem. Maybe it'll  be easier for you just to try the new npm version when it's released later this week.

**Comment by jbenzshawel** [comment]
@phongcao thanks this is resolved however I noticed some issues going from `0.9.1` to `0.9.3`. 

In my testing if I did not re-register for push notifications after installing this updated package the app crashes when a push notification is received - this happened both in the simulator and on a device. Could this be related to updating the Azure Notification Hubs SDK in PR94?

All of the configuration changes in the `AndroidManifest.xml` and `build.gradle` for moving from `0.9.1` to `0.9.3` were completed with this testing
